Find a functionally equivalent code:android.transition.Fade.createAnimation:COMMENT
Method Modifier: private     
Comment:/**
 * Utility method to handle creating and running the Animator.
 */

@@ -1,18 +1,14 @@
 {
     if (startAlpha == endAlpha) {
-        // run listener if we're noop'ing the animation, to get the end-state results now
-        if (listener != null) {
-            listener.onAnimationEnd(null);
-        }
         return null;
     }
-    final ObjectAnimator anim = ObjectAnimator.ofFloat(view, "transitionAlpha", startAlpha, endAlpha);
+    view.setTransitionAlpha(startAlpha);
+    final ObjectAnimator anim = ObjectAnimator.ofFloat(view, "transitionAlpha", endAlpha);
     if (DBG) {
         Log.d(LOG_TAG, "Created animator " + anim);
     }
-    if (listener != null) {
-        anim.addListener(listener);
-        anim.addPauseListener(listener);
-    }
+    FadeAnimatorListener listener = new FadeAnimatorListener(view);
+    anim.addListener(listener);
+    anim.addPauseListener(listener);
     return anim;
 }

